kingpin/actors/base.py

```diff
@@ -488,7 +488,10 @@ def execute(self):
             raise gen.Return()
 
         try:
-            result = yield self.timeout(self._execute)
+            if hasattr(self, '_execution_timeout'):
+                result = yield self._execution_timeout(self._execute)
```

Comment:
Wrapping the update-timeout checking at this level is problematic because it could actually lead to the stack update being canceled before the latest update is even tried. Here's the workflow:

1. Bob issues an stack update at 12:05PM and the stack goes into an UPDATING state...
2. Joe calls Kingpin to update the stack at 12:20PM with a 5M timeout.. but doesn't know anything about Bob doing that work.
3. Kingpin immediately starts a 5M timer and THEN checks the stack's current state (with the `ensure_state()` method). It sits on the stack while it finishes its UPDATING from Bob before it ever begins the update for Joe.
4. Bob's update takes a while and goes on to 12:25PM - at which point Joe's Kingpin instance issues a CANCEL call to the stack update.

See the issue?